sdk/ml/azure-ai-ml/azure/ai/ml/entities/_component/code.py

```diff
@@ -0,0 +1,75 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+import os
+from pathlib import Path
+from typing import Dict, List, Optional, Union
+
+from azure.ai.ml._utils._asset_utils import get_content_hash, get_ignore_file, IgnoreFile
+from azure.ai.ml._utils._pathspec import GitWildMatchPattern, normalize_file
+from azure.ai.ml.constants._common import ArmConstants
+from .._assets import Code
+
+
+class ComponentIgnoreFile(IgnoreFile):
+    _COMPONENT_CODE_IGNORES = ["__pycache__"]
+
+    def __init__(self, file_path: Optional[Union[str, Path]] = None):
+        super(ComponentIgnoreFile, self).__init__(file_path=file_path)
+        self._path_spec = self._create_pathspec()
+
+    @staticmethod
+    def from_ignore_file(ignore_file: IgnoreFile) -> "ComponentIgnoreFile":
+        return ComponentIgnoreFile(ignore_file.path)
+
+    def _create_pathspec(self) -> List[GitWildMatchPattern]:
+        """Override to add custom ignores."""
+        if not super(ComponentIgnoreFile, self).exists():
+            path_spec = []
+        else:
+            path_spec = super(ComponentIgnoreFile, self)._create_pathspec()
+        path_spec.extend([GitWildMatchPattern(ignore) for ignore in self._COMPONENT_CODE_IGNORES])
+        return path_spec
+
+    def is_file_excluded(self, file_path: Union[str, Path]) -> bool:
+        if self.exists():
+            return super(ComponentIgnoreFile, self).is_file_excluded(file_path)
+        file_path = str(file_path)
+        norm_file = normalize_file(file_path)
+        for pattern in self._path_spec:
+            if pattern.include is not None:
+                if norm_file in pattern.match((norm_file,)):
+                    return bool(pattern.include)
+        return False
+
+
+class ComponentCode(Code):
```

Comment:
this looks a little abundant for me as Code is a use-then-drop entity. Is there any concern to add a hidden init parameter `ignore_file` or `_ignore_file` to the base class?

Reply:
For the same reason, we may pass a InternalComponentIgnoreFile to original InternalCode instead of adding a specific class

Reply:
yes, that makes sense. I originally think it as a separate object, so I wrap it.
based on above discussion, I think it's more proper to add in base class.

Reply:
remove `ComponentCode` and `InternalCode`'s `__init__`, ignore file will be passed in the base class.